.github/workflows/lint-and-test.yml

```diff
@@ -0,0 +1,114 @@
+name: lint and test
+
+on: [push, pull_request]
+
+jobs:
+    build-conode:
+        name: build conode
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+                working-directory: conode
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-go@v2
+
+            - run: make docker_start_detached docker_wait
+            - run: docker exec conode sh -c './bcadmin latest --bc conodes/bc*cfg'
+
+    test-dynacred:
+        name: test dynacred
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+                working-directory: dynacred
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v2
+
+            - uses: actions/checkout@v2
+              with:
+                  repository: dedis/cothority
+                  path: cothority
+            - run: make docker
+              working-directory: cothority
+            - run: rm -rf ../cothority
+
+            - run: npm install
+            - run: npm run test
+
+    lint-webapp:
+        name: lint webapp
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+                working-directory: webapp
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v2
+
+            - run: npm install
+            - run: npm run lint
+
+    build-webapp:
+        name: build webapp
+        runs-on: ubuntu-latest
+        needs: lint-webapp
+        defaults:
+            run:
+                working-directory: webapp
+        steps:
+            - uses: actions/checkout@v2
+            - uses: actions/setup-node@v2
+
+            - run: npm install
```

Comment:
Shouldn't we use `npm ci`? I think I had bad experiences with `npm i`, which does also update the libraries. And I'm not sure the `package.json` is correctly written to survive `npm i` also in the future...

Reply:
Ho, I though it was only a quick way to remove the node_modules and install the pkgs, but [the documentation] states that it is designed for CI and behave differently.